utils/node_local_test.py

```diff
@@ -1946,10 +1963,125 @@ def test_rename(self):
         if dfuse.stop():
             self.fatal_errors = True
 
-        # Finally, perform some more I/O so we can tell from the dfuse logs where the test ends and
-        # dfuse teardown starts.  At this point file 1 and file 2 have been deleted.
-        time.sleep(1)
-        print(os.statvfs(self.dfuse.dir))
+    @needs_dfuse
+    def test_rename(self):
```

Comment:
don't see where you added a test for the changes you made in this PR for RENAME_NOREPLACE

Reply:
i see. it might be worth it to test some case where EXIST is actually returned because of this. anyway, seems sufficient for now as is with mv.

Reply:
I'm working adding  C test code for dfuse in https://github.com/daos-stack/daos/pull/7286 but it's slow progress right now.